reposerver/repository/repository.go

```diff
@@ -1707,13 +1745,35 @@ func (s *Service) newHelmClientResolveRevision(repo *v1alpha1.Repository, revisi
 	return helmClient, version.String(), nil
 }
 
+// directoryPermissionInitializer ensures the directory has read/write permissions and returns
+// a function that can be used to remove all permissions.
+func directoryPermissionInitializer(rootPath string) goio.Closer {
+	if _, err := os.Stat(rootPath); err == nil {
+		if err := os.Chmod(rootPath, 0755); err != nil {
```

Comment:
No. 0700 would be wrong, because it gives full access to the owner, and we want to prevent the owner from entering in the directory. So, all x bits on the directory must be gone :) I really would make it 0000, or 0644. (x actually is 1, w is 2 and r 4).

Reply:
Ah, it is the root path.

Reply:
It is cloned repo path. The owner actually needs full permissions there because we need to both read/write and execute binaries there to generate manifests. 
I changed group and other to 0 per Jesse suggestion because we have only one user.